<a href="https://colab.research.google.com/github/EliasT427/learning-Python/blob/main/Copy_of_1_printing_variables.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Instructions
Below is a list of elements where each element is a different data type. Your task is to perform some basic analysis on the list of elements to come up with the following report.

1. What is the data type of the element in position 3?
2. are the data types in position 0 and position 2 the same?
3. index the list so that you return the tuple. Assign the tuple to a new variable called `extracted`. What is the data type of the element in the 2nd position of this tuple?
4. Index the list to return the string `return me`, what are the characters at position 0, 3, and 5
5. Index the list to return the string `return me`, concatenate the characters at positions 0, 3, and 5 to create a new string. What is the new string?
6. return the element at position 9 and store it in a value called `squares`. What is the data type and what are it's values.
7. What is the result of the element at position 1 subtracted from the element at position 10
8. what is the memory address of the list `data`
9. are the memory addresses of the element in position 2 and position 8 the same? ***Bonus: If they are the same, explain why***

In [ ]:
data = [
    True,
    600.548,
    False,
    45j,
    [1,2,3,4,5],
    False,
    (True, "Minecraft is the best", 10.098, 'one billion', {1,2,3,4}),
    "return me",
    5 == 5,
    [x**2 for x in [1,2,3,4,5]],
    654.548    
]

In [ ]:
print(type(data[3]))
print(type(data[0]))
print(type(data[2]))
type(data[0]) == type(data[2])
extracted= data[6]
print(type(extracted[1]))
ReturnMe = data[7]
print(ReturnMe[0])
print(ReturnMe[3])
print(ReturnMe[5])
print(ReturnMe[0]+ReturnMe[3]+ReturnMe[5])
squares=data[8]
print(type(squares))
print(data[1]-data[10])
print(id(data))
id(data[2])==id(data[8])


<class 'complex'>
<class 'bool'>
<class 'bool'>
<class 'str'>
r
u
n
run
<class 'bool'>
-54.0
139929412922896


False

## YOUR ANSWERS GO HERE
***Double click on this cell to edit it***
1. Complex
2. yes they are the same
3. string
4. Run
5. Run
6. bool
7. -54 
8. 139929412922896
9. false 

